scm/github.go

```diff
@@ -466,27 +465,23 @@ func (s *GithubSCM) AddTeamMember(ctx context.Context, opt *TeamMembershipOption
 		}
 	}
 
+	var err error
 	if opt.TeamID < 1 {
-		team, _, err := s.client.Teams.GetTeamBySlug(ctx, opt.Organization, opt.TeamName)
-		if err != nil {
-			return ErrFailedSCM{
-				Method:   "AddTeamMember",
-				Message:  fmt.Sprintf("failed to get team %s in organization %s", opt.TeamName, opt.Organization),
-				GitError: err,
-			}
-		}
-		opt.TeamID = team.GetID()
+		_, _, err = s.client.Teams.AddTeamMembershipBySlug(ctx, opt.Organization, slug.Make(opt.TeamName), opt.Username,
+			&github.TeamAddTeamMembershipOptions{Role: opt.Role})
+	} else {
+		_, _, err = s.client.Teams.AddTeamMembershipByID(ctx, int64(opt.OrganizationID), int64(opt.TeamID), opt.Username,
+			&github.TeamAddTeamMembershipOptions{Role: opt.Role})
 	}
 
-	if _, _, err := s.client.Teams.AddTeamMembership(ctx, int64(opt.TeamID), opt.Username,
-		&github.TeamAddTeamMembershipOptions{Role: opt.Role}); err != nil {
-		return ErrFailedSCM{
+	if err != nil {
+		err = ErrFailedSCM{
 			GitError: err,
 			Method:   "AddTeamMember",
-			Message:  fmt.Sprintf("failed to add user %s to team ID %d with role %s", opt.Username, opt.TeamID, opt.Role),
+			Message:  fmt.Sprintf("failed to add user username: %s to team (ID %d, team name: %s) with role %s", opt.Username, opt.TeamID, opt.TeamName, opt.Role),
```

Comment:
user username sounds weird. I like the old error msg, maybe combined with the new one. Maybe add single quote or parens around the name, e.g. `... add user (meling) to team ...`. That is, `... user (%s) to team (ID %d, ...`

Reply:
Fixed.